.github/workflows/pull-request.yml

```diff
@@ -23,3 +23,20 @@ jobs:
 
       - name: run yamllint
         run: yamllint .github/workflows/*.yml
+  checkpatch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: install codespell
+        run: sudo apt update && sudo apt install -y codespell
+
+      - name: run codespell
+        run: curl ${{ github.event.pull_request.commits_url }} | jq '.[].sha'
```

Comment:
The curl is awkward. Can you try this instead? From https://github.com/actions/runner/issues/342#issuecomment-875823278

```yaml
with:
    COMMIT_RANGE: ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}

-name : checkpatch
  ./scripts/checkpatch.pl -g "${COMMIT_RANGE}" --codespell ...
```